packages/vc-verification/src/Ipfs-credential-resolver.ts

```diff
@@ -31,11 +31,30 @@ export class IpfsCredentialResolver implements CredentialResolver {
    * @returns
    */
   async getCredential(did: string, role: string) {
+    const credentials = await this.credentialsOf(did);
+    return credentials.find(
+      (claim) =>
+        claim.credentialSubject.role.namespace === role ||
+        utils.namehash(claim.credentialSubject.role.namespace) === role
```

Comment:
This is for the case when you fetch the `RoleDefinition` and the issuer type is `Role`. This role is namehash (since it is coming from ens contract). This namehash(role) becomes the role (to be queried) for the verification of next issuer. Thus in the next step the `namehash(role)` from ens and issuer `did` is the new params for the issuer verification.